pkg/controller/persistentvolume/persistentvolume_controller.go

```diff
@@ -0,0 +1,1438 @@
+// absence of this annotation means the binding was done by the user (i.e.
+// pre-bound).
+const annBoundByController = "pv.kubernetes.io/bound-by-controller"
+
+// annClass annotation represents a new field which instructs dynamic
+// provisioning to choose a particular storage class (aka profile).
+const annClass = "volume.alpha.kubernetes.io/storage-class"
+
+// This annotation is added to a PV that has been dynamically provisioned by
+// Kubernetes. It's value is name of volume plugin that created the volume.
+// It serves both user (to show where a PV comes from) and Kubernetes (to
+// recognize dynamically provisioned PVs in its decissions).
+const annDynamicallyProvisioned = "pv.kubernetes.io/provisioned-by"
+
+// Name of a tag attached to a real volume in cloud (e.g. AWS EBS or GCE PD)
+// with namespace of a persistent volume claim used to create this volume.
+const cloudVolumeCreatedForClaimNamespaceTag = "kubernetes.io/created-for/pvc/namespace"
+
+// Name of a tag attached to a real volume in cloud (e.g. AWS EBS or GCE PD)
+// with name of a persistent volume claim used to create this volume.
+const cloudVolumeCreatedForClaimNameTag = "kubernetes.io/created-for/pvc/name"
+
+// Name of a tag attached to a real volume in cloud (e.g. AWS EBS or GCE PD)
+// with name of appropriate Kubernetes persistent volume .
+const cloudVolumeCreatedForVolumeNameTag = "kubernetes.io/created-for/pv/name"
+
+// Number of retries when we create a PV object for a provisioned volume.
+const createProvisionedPVRetryCount = 5
+
+// Interval between retries when we create a PV object for a provisioned volume.
+const createProvisionedPVInterval = 10 * time.Second
+
+// PersistentVolumeController is a controller that synchronizes
+// PersistentVolumeClaims and PersistentVolumes.
+type PersistentVolumeController struct {
+	volumes          persistentVolumeOrderedIndex
+	volumeController *framework.Controller
+	claims           cache.Store
+	claimController  *framework.Controller
+	kubeClient       clientset.Interface
+	stopChannels     map[string]chan struct{}
+	eventRecorder    record.EventRecorder
+	cloud            cloudprovider.Interface
+	recyclePluginMgr vol.VolumePluginMgr
+	provisioner      vol.ProvisionableVolumePlugin
+	clusterName      string
+
+	// PersistentVolumeController keeps track of long running operations and
+	// makes sure it won't start the same operation twice in parallel.
+	// Each operation is identified by unique operationName.
+	// Simple keymutex.KeyMutex is not enough, we need to know what operations
+	// are in progress (so we don't schedule a new one) and keymutex.KeyMutex
+	// does not provide such functionality.
+
+	// runningOperationsMapLock guards access to runningOperations map
+	runningOperationsMapLock sync.Mutex
+	// runningOperations is map of running operations. The value does not
+	// matter, presence of a key is enough to consider an operation running.
+	runningOperations map[string]bool
```

Comment:
> GoRoutineMap is merged,

Not yet: https://github.com/kubernetes/kubernetes/pull/25400

I'll reuse it in separate PRs.